crates/workspace/src/pane.rs

```diff
@@ -2053,13 +2063,15 @@ impl Pane {
                 self.set_preview_item_id(None, cx);
             }
 
-            self.workspace
-                .update(cx, |_, cx| {
-                    cx.defer_in(window, move |_, window, cx| {
-                        move_item(&pane, &pane, id, destination_index, window, cx)
-                    });
-                })
-                .ok()?;
+            if ix != destination_index {
```

Comment:
Don't go through with a move if the destination is the same as its origin location.